arch/risc-v/src/common/riscv_debug.c

```diff
@@ -359,25 +325,95 @@ int up_debugpoint_add(int type, void *addr, size_t size,
 
   if (size > 1 && g_support_napot)
     {
-      mc.match = MATCH_TYPE_TOPBITS;
+      reg.match.match = MATCH_TYPE_TOPBITS;
       addr_napot = ((uintptr_t)addr & ~(size - 1)) |
                     ((size - 1) >> 1);
       WRITE_CSR(CSR_TDATA2, addr_napot);
     }
   else
     {
-      mc.match = MATCH_TYPE_EQUAL;
+      reg.match.match = MATCH_TYPE_EQUAL;
       WRITE_CSR(CSR_TDATA2, (uintptr_t)addr);
     }
 
+  return reg.reg;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_debugpoint_add
+ ****************************************************************************/
```

Comment:
Please include complete description, including parameters

Reply:
This function is a common function which have detaild description in https://github.com/apache/nuttx/blob/fe45d8aace2683b212e4ca2b9166a5df93c760fe/include/nuttx/arch.h#L2975-L2998

Maybe no need to dup it here?